samples/fuel_gauge/src/main.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <npmx_driver.h>
+
+#include "nrf_fuel_gauge.h"
+#include "fuel_gauge.h"
+
+#define LOG_MODULE_NAME main
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+/**
+ * @brief Callback function to be used when VBUSIN VOLTAGE event occurs.
+ *
+ * @param[in] p_pm Pointer to the instance of PMIC device.
+ * @param[in] type Callback type. Should be @ref NPMX_CALLBACK_TYPE_EVENT_VBUSIN_VOLTAGE.
+ * @param[in] mask Received event interrupt mask.
+ */
+void vbusin_voltage_callback(npmx_instance_t *p_pm, npmx_callback_type_t type, uint8_t mask)
+{
+	/* Current limit have to be applied each time when USB is (re)connected. */
+	if (mask & NPMX_EVENT_GROUP_VBUSIN_DETECTED_MASK) {
+		npmx_vbusin_task_trigger(npmx_vbusin_get(p_pm, 0),
+					 NPMX_VBUSIN_TASK_APPLY_CURRENT_LIMIT);
+	}
+}
+
+void main(void)
+{
+	const struct device *pmic_dev = DEVICE_DT_GET(DT_NODELABEL(npm_0));
+
+	if (!device_is_ready(pmic_dev)) {
+		LOG_ERR("PMIC device is not ready.");
+		return;
+	}
+
+	LOG_INF("PMIC device OK.");
+
+	/* Get pointer to npmx instance and other required instances. */
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+	npmx_charger_t *charger_instance = npmx_charger_get(npmx_instance, 0);
+	npmx_vbusin_t *vbusin_instance = npmx_vbusin_get(npmx_instance, 0);
+	npmx_adc_t *adc_instance = npmx_adc_get(npmx_instance, 0);
+
+	/* Register callback for VBUSIN VOLTAGE event. */
+	npmx_core_register_cb(npmx_instance, vbusin_voltage_callback,
+			      NPMX_CALLBACK_TYPE_EVENT_VBUSIN_VOLTAGE);
+
+	/* Enable detecting connected USB. */
+	npmx_core_event_interrupt_enable(npmx_instance, NPMX_EVENT_GROUP_VBUSIN_VOLTAGE,
+					 NPMX_EVENT_GROUP_VBUSIN_DETECTED_MASK);
+
+	/* Disable charger before changing charge current and termination voltage. */
+	npmx_charger_module_disable_set(charger_instance, NPMX_CHARGER_MODULE_CHARGER_MASK);
+
+	/* Set charging current. */
+	npmx_charger_charging_current_set(charger_instance, CONFIG_CHARGING_CURRENT);
+
+	/* Set maximum discharging current. */
+	npmx_charger_discharging_current_set(charger_instance, CONFIG_DISCHARGING_CURRENT);
```

Comment:
general note:
I've added `npmx_adc_max_charging_current_set()` and `npmx_adc_max_discharging_current_set()` functions to `npmx_adc.h` (they set the current limit values that are used in IBAT value calculations), you may want to use them somewhere here.

Reply:
No longer needed, values are set inside `npmx_charger_{dis}charging_current_set`